Improve server version logging info

Wolfsblvt opened the pull request:
Had a dummy idea. Just for people to get a quicker view on which version they are running, and why they may have issues.  
Also helps when someone provides output or screenshot.

- Capture commit date and print that next to the branch
- Info for being on a dev branch
- Info for not being on the latest commit (fetch should've gotten it, if update script was run)

![image](https://github.com/SillyTavern/SillyTavern/assets/9962104/a413b265-7845-4449-9f6e-9fb5b2f8d3fc)
![image](https://github.com/SillyTavern/SillyTavern/assets/9962104/e1c5354e-8284-4e82-9f9f-7042f771d432)
![image](https://github.com/SillyTavern/SillyTavern/assets/9962104/290de9bd-7343-4335-98dd-c8fe1071ee20)

Wolfsblvt committed:
Improve server version logging info

- Capture commit date and print that next to the branch
- Info for being on a dev branch
- Info for not being on the latest commit (fetch should've gotten it, if update script was run)

Cohee1207 commented:
I like everything beside the dev branch part. If someone got there already, it's no ordinary user so likely they know what they're doing anyway.

Wolfsblvt commented:
We (on the Discord) regularly tell people to test something out by getting the PR branch.  
That's why I thought it might be nice to *remind* them that they are not on staging anymore.

But I can see that being annoying. We should still skip the latest check if it's not `release` or `staging`. I can just remove the dev logging block and use the negated if.

Cohee1207 commented on src/util.js:
```diff
@@ -73,27 +73,37 @@ function getBasicAuthHeader(auth) {
+
+            // Might fail, but exception is caught. Just don't run anything relevant after in this block...
+            const localLatest = await git.cwd(cwd).revparse(['HEAD']);
+            const remoteLatest = await git.cwd(cwd).revparse([`origin/${gitBranch}`]);
```
You can easily get the name of tracking upstream branch with `const upstreamBranch = await git.cwd(process.cwd()).revparse(['--abbrev-ref', '@{u}']);`

Equivalent to cmd command: `git rev-parse --abbrev-ref @{u}`

Wolfsblvt replied:
What would getting the name of the upstream branch help us here? It only applies to forks, doesn't it?

I know I can get `upstream`. But then we'd have to make sure that upstream was fetched as well. The normal `UpdateAndStart.bat` will only do a `git pull`, which doesn't update upstream branches by default.

That's why I just thought like.... lets not log too much for actual forks, that'd be hard to handle for just a small info script.  
I mean I did similar "magic" with my `UpdateForkAndStart.bat`, but I don't think adding actual fetches inside the server script is a good idea.  
I was just leaning on what the update scripts already provided with the local repo.  
And if you ran the update fork script where you have an upstream it'll already stash, pull and all.

So, just that I get this right: Just removing the dev logging then?

Cohee1207 replied:
You're getting rid of hardcoded `origin` string. Because sometimes people not have it named `origin`, and it actually shows the tracking remote branch of the current local copy.

* Yes, by upstream I mean the remote branch

Cohee1207 replied:
> So, just that I get this right: Just removing the dev logging then?

Yes for dev logging.

Wolfsblvt replied:
Makes sense. it's the "remote tracking branch". git terminology :D

I also removed the dev logging as you requested.

Wolfsblvt committed:
Server logging utilize tracking branch

- Use tracking branch instead of hardcoded "origin"
- Remove dev logging message if not on "staging" or "release"

Cohee1207 committed:
Fix tpyo + add clarity + lint
